### PR TITLE
Global nav i4 - add selected global tab styles 

### PR DESCRIPTION
**Pull request by Addison-Stavlo**
Related to # Automattic/wp-calypso#93038 Automattic/wp-calypso#93037  Automattic/wp-calypso#93039

## Proposed Changes

Applies the styles outlined in the issues above.
* Special selected styles for sites dashboard, me, and reader tabs in global masterbar.
* Updates global masterbar text/icon colors and submenu background colors as outlined in Automattic/wp-calypso#93039
* Adds a small amount of top padding for the content frame back to mobile, to support the tab styles as recommended in Automattic/wp-calypso#93038

BEFORE
new-global-masterbar-styles-before

AFTER
new-global-masterbar-after

## Why are these changes being made?

* Per design request, iteration to make new interfaces look better in global area, and pe7F0s-1XD-p2

## Testing Instructions

* Run this branch of calypso.
* Smoke test the new styles in the global masterbar.
* Verify there are no regressions on the site level masterbar like my home.
* Test mobile/tablet viewports, and with various browsers.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/layout/masterbar/item-border-wrapper.jsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export default function ItemBorderWrapper( { isActive } ) {
```
Comment (Addison-Stavlo) [inline client/layout/masterbar/item-border-wrapper.jsx]: This initially was a wrapper that passed children. The name and classnames seem less fitting now that this is added interior to the Item component and not wrapping anything.

File: client/layout/masterbar/style.scss [hunk 2]
```diff
@@ -1401,3 +1414,53 @@ a.masterbar__quick-language-switcher {
 		opacity: 1;
 	}
 }
+
+.masterbar__item.has-global-border .item-border-wrapper {
```
Comment (Addison-Stavlo) [inline client/layout/masterbar/style.scss]: All of this is what allows us to position and shape elements for that awkward outward border radius/curve on the bottom of the selected global items. Maybe there is a less complex way I am unaware of, but simply leveraging pseudo elements on existing items didn't seem to give us enough control.

File: client/my-sites/sidebar/style.scss [hunk 3]
```diff
@@ -1122,7 +1122,7 @@ $font-size: rem(14px);
 		}
 
 		.focus-content:not(.has-no-sidebar):not(.has-no-masterbar) .layout__content {
-			padding-top: 46px;
+			padding-top: 62px;
```
Comment (Addison-Stavlo) [inline client/my-sites/sidebar/style.scss:1132]: Per https://github.com/Automattic/wp-calypso/issues/93038#issuecomment-2259029880

File: client/layout/masterbar/style.scss [hunk 4]
```diff
@@ -1401,3 +1414,53 @@ a.masterbar__quick-language-switcher {
 		opacity: 1;
 	}
 }
+
```
Comment (lupus2k) [inline client/layout/masterbar/style.scss]:
```suggestion
.is-global-sidebar-visible .masterbar__item {
	transition: none;
}

```
Removing transitions to `masterbar__item` looks like this:

https://github.com/user-attachments/assets/5a9e65c2-95b2-427a-8e2b-27103c0c4ecc

File: client/layout/masterbar/style.scss [hunk 5]
```diff
@@ -1401,3 +1414,53 @@ a.masterbar__quick-language-switcher {
...
+		.item-border-wrapper__left,
+		.item-border-wrapper__right {
+			display: block;
+			position: absolute;
+			height: var(--masterbar-item-active-border-radius);
+			width: var(--masterbar-item-active-border-radius);
+			background-color: var(--color-sidebar-background);
+			bottom: 0;
+			z-index: 1;
```
Comment (lupus2k) [inline client/layout/masterbar/style.scss]:
```suggestion
	display: block;
	opacity: 0;

	.is-global-sidebar-visible &.is-active {
		opacity: 1;
		top: 0;
		bottom: 0;
		right: 0;
		left: 0;

		.item-border-wrapper__left,
		.item-border-wrapper__right {
			display: block;
			position: absolute;
			height: var(--masterbar-item-active-border-radius);
			width: var(--masterbar-item-active-border-radius);
			background-color: var(--color-sidebar-background);
			bottom: 0;
			z-index: -1;
			transition: all 0.2s ease-in-out;
```
This is how it looks (first part of the video with animation at 10%, and then 100%)

https://github.com/user-attachments/assets/e9e0fcf9-543a-488c-9cdd-b41e99087b3f

Comment (Addison-Stavlo) [reply client/layout/masterbar/style.scss]: Thanks for exploring this! I pushed a commit based on what you explored and some things you mentioned in slack. Its definitely a bit better than it was now!

I am also going to try again to get this same outward curve effect with less elements, as that would probably have less potential visual conflicts like this.

Comment (Addison-Stavlo) [reply client/layout/masterbar/style.scss]: > to try again to get this same outward curve effect with less elements, as that would probably have less potential visual conflicts like this.

that ended up working out! we can cut pseudo elements with a mask instead, removing the elements that were making the transition funny.

**Comment by lupus2k** [comment]
It looks great with the latest changes even with the animations at 10% speed

https://github.com/user-attachments/assets/5dd45930-2fce-480c-9d15-3d97ac2e4ef2

**Review by eoigal — APPROVED** [review]
Works well for me and looks good. My only nitpick was do you think the hover should have the same border radius? I don't think its important, just something I noticed. 

Screenshot 2024-07-31 at 20 28 02

**Review by allilevine — COMMENTED** [review]
I tested in Brave, Firefox, and Safari, and at different screen sizes. In Brave's device toolbar views I noticed there's a line showing up under the tab:

Screen Shot 2024-07-31 at 3 19 39 PM
Screen Shot 2024-07-31 at 3 12 23 PM

That could be a selected state though. I also found the submenu background color strange, but I see that was the recommendation in https://github.com/Automattic/wp-calypso/issues/93039

Screen Shot 2024-07-31 at 3 45 14 PM

Otherwise it looks great! 🙌

**Comment by Addison-Stavlo** [comment]
>  In Brave's device toolbar views I noticed there's a line showing up under the tab

I also notice this in the chrome dev tools if I adjust the zoom level away from 100%, but only using the dev tools zoom. If I close dev tools and zoom in/out with cmd+plus I don't see the issue. Im wondering if this is a bug with the previewers or if we are overlooking something?

**Review by lupus2k — APPROVED** [review]
LGTM in all screen sizes!

**Comment by Addison-Stavlo** [comment]
> do you think the hover should have the same border radius?

I think its ok without it, as it wouldn't make sense for many of the regular items.

**Comment by youknowriad** [comment]
Hi There! where can I read more about this change. At first sight, this change felt like a regression for me. From a consistent sidebar across all the dashboards, we now have an inconsistent experience. At least that's how I felt when using .com

**Comment by davemart-in** [comment]
@youknowriad I'll DM you some links.

**Comment by Addison-Stavlo** [comment]
> From a consistent sidebar across all the dashboards, we now have an inconsistent experience.

consistently inconsistent 😅 I assume you meant "admin bar" and not "sidebar" given the context of this PR?

Whats consistent - should be the admin bar and sidebar across the surface area of a site. inside and outside calypso.

what should be consistently inconsistent -  is when we are outside the context of any particular site, like these global areas. The admin bar should have the same familiar items, but have subtle differences to indicate those global levels. The admin bar is not the only thing that does this, but the sidebar as well.

**Comment by youknowriad** [comment]
> what should be consistently inconsistent - is when we are outside the context of any particular site, like these global areas. The admin bar should have the same familiar items, but have subtle differences to indicate those global levels. The admin bar is not the only thing that has separate styles, this is also true for the global sidebar and global content areas as well.

Yes, I guess for me, being "global" or "local" doesn't feel enough to change the style of either the sidebar or admin bar. But I was focused solely on the admin bar for my feedback here.

**Comment by Addison-Stavlo** [comment]
> Yes, I guess for me, being "global" or "local" doesn't feel enough to change the style of either the sidebar or admin bar. But I was focused solely on the admin bar for my feedback here.

Makes sense, as it definitely could feel out of place if having other expectations. The separate styles for global areas have been iterated through the nav redesign and untangling design and development efforts this year. @lucasmendes-design is also a good point of contact for the design side here.

**Comment by davemart-in** [comment]
One related note: We ran a whole bunch of usability tests in the process of doing the original redesign. I'll send you some links via Slack. One of the big takeaways was that if the global screens look too similar to wp-admin that beginner users tend to get confused. They'd for instance think they were in wp-admin when in fact they were in the global state. By having a visual separation, we seemed to be able to largely resolve that issue.

**Pull request merged**